airflow/models/dag.py

```diff
@@ -1847,7 +1847,10 @@ def bulk_write_to_db(cls, dags: Collection["DAG"], session=None):
         for missing_dag_id in missing_dag_ids:
             orm_dag = DagModel(dag_id=missing_dag_id)
             dag = dag_by_ids[missing_dag_id]
-            if dag.is_paused_upon_creation is not None:
+
+            if dag.is_subdag:
+                orm_dag.is_paused = dag.parent_dag.get_is_paused()
+            elif dag.is_paused_upon_creation is not None:
```

Comment:
This feels a bit hacky to me. Can the subdag not be created with `is_paused_upon_creation` correctly set to True? Why not?

Reply:
@uranusjr Per my understanding, the `is_subdag` parameter is set to true only in the `_bag_dag` method, and setting `is_paused_upon_creation` there will impact existing subdags as well(don't think that will negatively impact but still). Changing is_paused there will be overwritten by this method for newly added ones. So, I chose to make the changes here. Please correct me if my understanding is wrong.
https://github.com/apache/airflow/blob/28e285ef9a4702b3babf6ed3c094af07c017581f/airflow/models/dagbag.py#L447-L450

Reply:
@uranusjr - any thoughts?

Reply:
> setting is_paused_upon_creation there will impact existing subdags as well (don't think that will negatively impact but still).

I’m definitely missing something, because I thought this is what this PR is for? When a parent DAG is set to pause, its subdags should also be paused?

Reply:
@uranusjr - Okay. So what is happening is, let's say you create a dag and deploy it and it is running. At a later point, you want to add an enhancement to the dag which is a subdag. You would expect the subdag to run in the next schedule. However, what is happening is, the newly added subdag is created in off state because `is_paused_upon_creation` by default is False. My PR fixes this issue by checking the status of the parent dag for such cases only. 

The workaround for this currently is, turn off the parent dag and turn it back on so that both the parent and subdag is in the same state.

> When a parent DAG is set to pause, its subdags should also be paused?

This is currently working fine and this PR does not change anything wrt this.
If you still have questions, I can ping you over the slack channel. Thanks!

Reply:
I’m still not following so please bear with me…

> However, what is happening is, the newly added subdag is created in off state because is_paused_upon_creation by default is False.

If the default is False (i.e. not paused), why is the subdag created in the off state?

The default `dags_are_paused_at_creation` value is actually True, so by default, dags are created in in the paused state. So when the parent DAG is unpaused, the newly-created subdag would indeed be in a different state. Is this what you’re trying to say? (but not describing the true/false on/off values correctly… which is understandable, the various negative value names are difficult to keep straight)

----

Assuming that’s the case, setting a subdag to unpaused by default if the parent dag is unpaused does sound like a reasonable thing to do, and the patch is also right (from what I can tell). But that would arguably be a backward incompatibility behavioural change I’m not sure we should just do. I hope another maintainer can provide second opinion on this.

Reply:
> If the default is False (i.e. not paused), why is the subdag created in the off state?

My bad, I just fixed my comment. It is true by default.

> The default dags_are_paused_at_creation value is actually True, so by default, dags are created in in the paused state. So when the parent DAG is unpaused, the newly-created subdag would indeed be in a different state. Is this what you’re trying to say? (but not describing the true/false on/off values correctly… which is understandable, the various negative value names are difficult to keep straight)

Yes right.